app/main.py

```diff
@@ -1,3 +1,72 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        else:
+            return Distance(
+                km=self.km + other
+            )
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+
+        return self
+
+    def __mul__(self, other: int) -> Distance:
+        return Distance(
+            km=self.km * other
+        )
+
+    def __truediv__(self, other: Any) -> Distance:
+        if other > 0:
+            return Distance(km=round(self.km / other, 2))
+        if other == 0:
+            raise ZeroDivisionError("You cannot divide by zero")
```

Comment:
The check for division by zero should be performed before the division operation. Consider rearranging the logic to check if `other` is zero first, and then proceed with the division if it is not.